packages/shared/src/hooks/useVotePost.ts

```diff
@@ -110,6 +112,7 @@ const useVotePost = <T extends { id: string } = { id: string }>({
       mutationKey: downvotePostMutationKey,
       onMutate: onDownvotePostMutate,
       onError: (err, _, rollback) => rollback?.(),
+      onSuccess: onDownvoted,
```

Comment:
I understand why is it here, Lee wanted to make sure the server side call actually passed before offering user blocking options.

That said, since we do have optimistic UI here I think we might want to just show the blocking UI `onMutate` since there could be a delay until server call and user could already navigate elsewhere. Downvote and blocking are not connected so no harm in having one or the other. So I would say we do not need `onDownvoted` here currently.

Reply:
> This is kind of what the onMutate handles no?

Yes, just to make sure it has been completed.

Ante's comment makes sense. @rebelchris WDYT about it?

Reply:
I think we always favor optimistic UI, so would put it all into the mutation handler.
We should aim to trust our calls will succeed basically.